----------------------------------------------------------------------
r12627 (orig r705):  clkao | 2004-10-19T03:57:28.850381Z

Give the base revision when opening files for propset.
This solves the ps regression for svn 1.0.x.

----------------------------------------------------------------------
--- SVK-0.23/lib/SVK/Command/Propset.pm  (revision 12626)
+++ SVK-0.23-patched/lib/SVK/Command/Propset.pm  (revision 12627)
@@ -31,15 +31,15 @@ sub lock {
 sub do_propset_direct {
     my ($self, $target, $propname, $propvalue) = @_;
 
-    my $fs = $target->{repos}->fs;
     if ($self->{revprop}) {
+	my $fs = $target->{repos}->fs;
         my $rev = (defined($self->{rev}) ? $self->{rev} : $target->{revision});
         $fs->change_rev_prop ($rev, $propname => $propvalue);
         print loc("Property '%1' set on repository revision %2.\n", $propname, $rev);
         return;
     }
 
-    my $root = $fs->revision_root ($fs->youngest_rev);
+    my $root = $target->root;
     my $kind = $root->check_path ($target->path);
 
     die loc("path %1 does not exist.\n", $target->path) if $kind == $SVN::Node::none;
@@ -53,13 +53,13 @@ sub do_propset_direct {
 	    $editor->change_dir_prop ($editor->{_root_baton}, $propname, $propvalue);
 	}
 	else {
-	    my $baton = $editor->open_directory ($path, 0, -1);
+	    my $baton = $editor->open_directory ($path, 0, $target->{revision});
 	    $editor->change_dir_prop ($baton, $propname, $propvalue);
 	    $editor->close_directory ($baton);
 	}
     }
     else {
-	my $baton = $editor->open_file ($path, 0, -1);
+	my $baton = $editor->open_file ($path, 0, $target->{revision});
 	$editor->change_file_prop ($baton, $propname, $propvalue);
 	$editor->close_file ($baton, undef);
     }
